### PR TITLE
split Server into Listener and ServerState

### PR DESCRIPTION
**Pull request by Akuli**
We aren't sure yet whether any of this is a good idea.

### CODE REVIEW

File: mantatail.py [hunk 1]
```diff
@@ -2,92 +2,93 @@
...
+                    else:
+                        call_handler_function(state, user, message)
 
-                        if user.closed_connection:
-                            return
+                    if user.closed_connection:
+                        return
 
 
 class User:
```
Comment (ThePhilgrim) [inline mantatail.py:94]:
```suggestion
class UserConnection:
```

I think it makes sense to rename this in this refactor to clarify its responsibility

File: mantatail.py [hunk 2]
```diff
@@ -2,92 +2,93 @@
...
-                            # ex. "command.handle_nick" or "command.handle_join"
-                            call_handler_function = getattr(command, parsed_command)
-                        except AttributeError:
-                            command.error_unknown_command(user, verb_lower)
-                        else:
-                            call_handler_function(self, user, message)
+                        user_socket.sendall(command.error_not_registered())
+
+                    if _user_message and _nick:
+                        user = User(user_host, user_socket, _user_message, _nick)
```
Comment (ThePhilgrim) [inline mantatail.py:78]:
```suggestion
                        user = UserConnection(user_host, user_socket, _user_message, _nick)
```

**Comment by ThePhilgrim** [comment]
Can we update `send_string()` to `send_string_to_client()` or do you think that's a bad idea?

**Review by ThePhilgrim — CHANGES_REQUESTED** [review]

File: command.py [hunk 3]
```diff
@@ -52,7 +56,7 @@ def handle_join(server: mantatail.Server, user: mantatail.User, channel_name: st
         #   * Send End of Name list (366)
 
         # TODO: Check for:
-        #   * User invited to channel
+        #   * UserConnection invited to channel
```
Comment (Akuli) [inline command.py:59]:
```suggestion
        #   * User invited to channel
```

**Review by ThePhilgrim — APPROVED** [review]

**Pull request merged**